core/history_api.php

```diff
@@ -238,6 +285,31 @@ function history_get_range_result_filter( $p_filter, $p_start_time = null, $p_en
 }
 
 /**
+ * Creates and executes a query for the history rows related to bugs matched by the provided filter
+ * @param  array $p_filter           Filter array
+ * @param  integer $p_start_time     The start time to filter by, or null for all.
+ * @param  integer $p_end_time       The end time to filter by, or null for all.
+ * @param  string  $p_history_order  The sort order.
+ * @return database result to pass into history_get_event_from_row().
+ */
+function history_get_range_result_filter( $p_filter, $p_start_time = null, $p_end_time = null, $p_history_order = null ) {
+	$t_query_options = array();
+	if ( $p_history_order !== null ) {
+		$t_query_options['order'] = $p_history_order;
+	}
+	if ( $p_start_time !== null ) {
+		$t_query_options['start_time'] = $p_start_time;
+	}
+	if ( $p_end_time !== null ) {
+		$t_query_options['end_time'] = $p_end_time;
+	}
+	if ( $p_filter !== null ) {
+		$t_query_options['filter'] = $p_filter;
+	}
+	return history_query_result( $t_query_options );
+}
+
+/**
  * Creates and executes a query for the history rows matching the specified criteria.
  * @param  integer $p_bug_id         The bug id or null for matching any bug.
```

Comment:
I opt to not change expected parameters in this function.
If an array of bugs are requested, using the new history_query_result() should be better.